locksmith/src/controllers/v2/metadataController.ts

```diff
@@ -246,3 +247,21 @@ export const updateUsersMetadata: RequestHandler = async (
     error,
   })
 }
+
+export const readUserMetadata: RequestHandler = async (request, response) => {
+  const userAddress = Normalizer.ethereumAddress(request.params.userAddress)
+  const tokenAddress = Normalizer.ethereumAddress(request.params.lockAddress)
+  const network = Number(request.params.network)
+  const user = await getMetadata(
+    tokenAddress,
+    userAddress,
+    true /* includeProtected */
```

Comment:
I think this sounds be true only if the connected user is the same as "userAddress".

Reply:
> Sorry, do we want to return any public data if the user address and connected user is different? the modal data seems more like a payment info so I assumed to not return any data if they are different.

Yes!